ros2bag/ros2bag/verb/record.py

```diff
@@ -55,6 +62,15 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
                   'Default it is zero, recording written in single bagfile and splitting '
                   'is disabled.'
         )
+        parser.add_argument(
+            '-c', '--compression-mode', type=str, default=CompressionMode.NONE,
+            choices=[mode.value for mode in CompressionMode],
+            help='Determine whether to compress by file or message. Default is "NONE".'
```

Comment:
Design nits:
* Arguments are lower case in GNU programs.
* I think CompressionMode should default to the `compression-format`'s default `compression-mode`.
* Additionally, I think the long option should only exist and repurpose the short option `-c` to select a default `compression-mode` and `compression-format`.